### DISCUSSION TITLE
Exporting smart playlists for use on other computer

### DISCUSSION THREAD

**Discussion by JOJ0** [beetbox/beets#4373] [Q&A] [answered]
Hi,
I realised that out-of-the-box what I want to is not possible but I might be missing something. Before I start scripting, I'd like to ask here:

- I have a couple of smart playlists generated/configured using the spl plugin (m3u files)
- I'd like to copy those playlists together with it's corresponding media files to an external flashdrive
- And be able to put it in any computer an play the playlists.

What I tried:
- Installed the convert plugin, set it to "never convert lossles" so it just copies all mp3 files to a destination folder.
- The plugin copies files to the structure given in the beets global path: configuration.
- How would I now copy and rewrite my m3u files with the new correct paths to the files on the flashdrive?

Maybe there is an entirely different approach I'm not seeing at the moment. Thanks for any hints and creative ideas around the topic! :-)

**Comment by JOJ0** [comment]
What I figured out by now is that path: can be set separately by the convert plugin so I can easily copy stuff to the destination folder in a "flat" format / all in one folder, which is a good basis for m3u playlists with relative paths!

**Comment by arsaboo** [comment]
This may not be directly related to your original question, but I have all my music on a NAS drive and I mount it to `/data/music` on every device. 

It would be nice to have more robust playlist features - importing, syncing, etc.

**Reply by JOJ0** [reply]
Have you tried the "playlist" and "smartplaylist" plugins?

**Reply by RollingStar** [reply]
caliope is also distantly related  https://discourse.beets.io/t/calliope-antisocial-music-recommendations/1532

**Comment by JOJ0** [comment]
What I have so far: 

```
(beets) ➜  beets git:(my_fixes) ✗ for m3u in /Users/jojo/Music/Techno/0-playlists/*; do echo beet convert -y playlist:$m3u --playlist "~/Music/Techno/0-export/${m3u##*/}"; done
beet convert -y playlist:/Users/jojo/Music/Techno/0-playlists/110..130_TechnoJungleDub_Hi.m3u --playlist ~/Music/Techno/0-export/110..130_TechnoJungleDub_Hi.m3u
beet convert -y playlist:/Users/jojo/Music/Techno/0-playlists/110..130_TechnoJungleDub_Lo.m3u --playlist ~/Music/Techno/0-export/110..130_TechnoJungleDub_Lo.m3u
beet convert -y playlist:/Users/jojo/Music/Techno/0-playlists/110..130_UK-BreaksTechHardcore_Hi.m3u --playlist ~/Music/Techno/0-export/110..130_UK-BreaksTechHardcore_Hi.m3u
beet convert -y playlist:/Users/jojo/Music/Techno/0-playlists/110..130_UK-BreaksTechHardcore_Lo.m3u --playlist ~/Music/Techno/0-export/110..130_UK-BreaksTechHardcore_Lo.m3u
beet convert -y playlist:/Users/jojo/Music/Techno/0-playlists/115..126_House.m3u --playlist ~/Music/Techno/0-export/115..126_House.m3u
beet convert -y playlist:/Users/jojo/Music/Techno/0-playlists/124..135_Electro.m3u --playlist ~/Music/Techno/0-export/124..135_Electro.m3u
...
...
```


- To access the files in my generated smartplaylist files I use the "playlist" plugin (that enables the query-syntax `playlist:playlist_path`)
- The convert plugin copies to the configured `dest:` dir in a configured `path:` format.
- What's missing: A feature in the convert plugin that would save a m3u playlist to the file given in --playlist. I'm starting to quick and dirty implement this for now and hope I can contribute it as a proper new feature to the convert plugin when I find more time for it.

**Comment by JOJ0** [comment, marked as answer]
There is a new beets feature in the queue that will implement this functionality once it gets merged: https://github.com/beetbox/beets/pull/4399
